### PR TITLE
pep8ify backends.

### PR DESCRIPTION
**Pull request by anntzer**
Some (clearly intended as internal) globals were removed without
deprecation as they cannot be properly deprecated; see changelog.

PIXELS_PER_INCH is not used anywhere.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backends/backend_pgf.py [hunk 1]
```diff
@@ -76,33 +76,30 @@ def get_preamble():
...
 NO_ESCAPE = r"(?<!\\)(?:\\\\)*"
 re_mathsep = re.compile(NO_ESCAPE + r"\$")
-re_escapetext = re.compile(NO_ESCAPE + "([_^$%])")
-repl_escapetext = lambda m: "\\" + m.group(1)
-re_mathdefault = re.compile(NO_ESCAPE + r"(\\mathdefault)")
-repl_mathdefault = lambda m: m.group(0)[:-len(m.group(1))]
+_replace_escapetext = functools.partial(
+    re.compile(NO_ESCAPE + "(?=[_^$%])").sub, lambda m: "\\")
+_replace_mathdefault = functools.partial(
+    re.compile(NO_ESCAPE + r"(\\mathdefault)").sub, lambda m: "")
```
Comment (anntzer) [inline lib/matplotlib/backends/backend_pgf.py]: I think these regex substitutions are simpler to read: they simply mean, in the first case: "when the next character is a _, ^, $ or %" (not preceded by an escape), insert a backslash" and in the second case: "replace \mathdefault (not preceded by an escape) by the empty string".

File: doc/api/next_api_changes/2019-04-21-AL.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Removals
+````````
+
+The following API elements have been removed:
+- ``backend_gtk3.PIXELS_PER_INCH``,
+- ``backend_pgf.re_escapetext``, ``backend_pgf.repl_escapetext``,
+  ``backend_pgf.re_mathdefault``, ``backend_pgf.repl_mathdefault``,
```
Comment (timhoffm) [inline doc/api/next_api_changes/2019-04-21-AL.rst]: /home/circleci/project/doc/api/next_api_changes/2019-04-21-AL.rst:7: WARNING: Unexpected indentation.

likely causes the doc build failure.

Comment (anntzer) [reply doc/api/next_api_changes/2019-04-21-AL.rst]: thanks, should be fixed now

File: lib/matplotlib/backends/backend_pgf.py (hunk 1 above)
Comment (efiring) [reply lib/matplotlib/backends/backend_pgf.py]: I suggest you take this opportunity to put your English translations in as comments.  The old and new versions are incomprehensible to me, and I suspect to many others, without extensive research.  I have no objection to your new versions, though.

Comment (anntzer) [reply lib/matplotlib/backends/backend_pgf.py]: done

**Pull request merged**